### DISCUSSION TITLE
Poco F3 and occasional disconnections with Dexcom G6 transmitter

### DISCUSSION THREAD

**Discussion by dar-kod** [NightscoutFoundation/xDrip#1989] [Q&A]
Phone: Poco F3 (android 11)
xDrip+: 92b6d8d-2022.02.09

With Poco F3 I have seen that from time to time there is a disconnection with transmitter and no values come to xDrip+. I usually do something like this: I change the last number of Dexcom Transmitter ID in the settings and then change ID to correct one. It triggers xDrip+ to reconnect with transmitter. But I have to wait that it asks to pair (it is tricky, sometimes you can click a wrong button) and it asks again to connect (what is typical for Xiaomi). Do you know how to stop Xiaomi to kill Bluetooth connection? Bluetooth in Samsung for example can be set to be not optimized for battery saving, but here there is no such option.

**Comment by Navid200** [comment]
Can you share the xDrip logs for the moment a disconnect starts?
When you look at xDrip logs, you can scroll up or down to go back and forth in time.  So, the next time there is a disconnect, please note the time of the last reading on screen.  Then, view the logs and go back until you get to logs a few minutes before the last reading and capture a screenshot and share with us.

Please note that xDrip logs may contain IDs or passwords.  So, please backout such before sharing.

**Comment by dar-kod** [comment]
I found the option in `Settings/Privacy/Manage/Special app access/Battery optimisation`. Switch to `All apps`. Click on `Bluetooth/Don't optimise`.

I will see how it works and send the logs, if the problem persists.

**Comment by dar-kod** [comment]
@Navid200, I have just noticed disconnection again, so it is clearly not something connected with battery optimization. In this disconnection time I was in another room maybe 5-6 meters away from the phone, on which distance usually I would have connection. I noticed it at 0:15 (Signal Missed 47 Minutes ago) and made screenshots of the main screen and System Status. Then I reconnected with the method I described above. Here below are two screenshots of logs. Signal started to miss at 23:28.

![photo_2022-02-14_00-56-30](https://user-images.githubusercontent.com/24446806/153781516-65f448fd-5fa5-4d15-b155-b4049b4f455f.jpg)
![photo_2022-02-14_00-48-35](https://user-images.githubusercontent.com/24446806/153781499-706dacf5-9c40-41b1-b5a3-447f722d135c.jpg)

**Reply by Navid200** [reply]
Can you show a screenshot of the page:
Settings -> G5/G6 Debug Settings

And are the settings on the page as they were when the disconnect and logs you are showing occured?

**Comment by dar-kod** [comment]
When the disconnect and logs occurred the G5/G6 Debug Settings were the same as these down:

![Screenshot_2022-02-14-08-56-25-795_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/24446806/153853735-8f0e4ac0-007d-466a-9283-2ab69d7b5e58.jpg)
![Screenshot_2022-02-14-08-57-15-672_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/24446806/153853752-f7285189-5591-40dc-bdd2-3ab97def78f0.jpg)
![Screenshot_2022-02-14-08-57-30-303_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/24446806/153853796-fb4898f2-3068-491a-bfda-b645e5be9ff6.jpg)

**Reply by Navid200** [reply]
You can leave all of that as is.  But, disable "Allow OB1 Unbonding" as a test to see if it makes any change.

**Comment by dar-kod** [comment]
I disabled "Allow OB1 Unbonding" now. I think I forgot to tell that this usually occurs after I am away from the phone for 15-20 minutes and sometimes it reconnects, and sometimes I do myself.

**Comment by Navid200** [comment]
Please try the latest Nightly if you are still having connectivity problems:
https://github.com/NightscoutFoundation/xDrip/releases/tag/2022.06.09

**Reply by dar-kod** [reply]
I have already installed the latest Nightly. Thank you for the information. I have still disabled "Allow OB1 Unbonding" but I cannot remember if I was enough long away from the phone. I didn't have disconnections. Furthermore, I may try to test it. Should I enable "Allow OB1 Unbonding"?

**Reply by Navid200** [reply]
To begin with, if I were you, I would enable it.  If enabling it results in disconnects, you can then disable it.

But, please post here and tell us what happens either way.

**Reply by dar-kod** [reply]
Thank you. I did as you wrote. Yesterday I was a few times away from the phone for about 30 minutes or more and when I came to the phone the connection would be restored automatically.

**Reply by dar-kod** [reply]
@Navid200, I experienced today 2 times very stage behavior. This second time I see missed signal before 78 minutes, and then it connected. I was not far away from the phone. I was around. The similar happened first time, but it was something like missed signal before  37 minutes, when I have seen it, and it reconnected after that. I will enable "Allow OB1 Unbonding" now.

**Reply by dar-kod** [reply]
Sorry, if I remembered good I disabled "Allow OB1 Unbonding" but I see it was enabled. I will disable to see if it will make change.

**Reply by dar-kod** [reply]
I got again long disconnection and had to change Transmitter ID as I described in my first writing here.

**Reply by dar-kod** [reply]
I have now disconnection for 81 minutes and phone is near me. What to do?

**Reply by dar-kod** [reply]
I rebooted the phone, and it seems to work now good. I will post an update, if disconnection happens again.

**Comment by dar-kod** [comment]
After update to MIUI  Global 13.0.4 yesterday, I have constant disconnections. Reboot of phone doesn't help. It seems that Android has to learn again about this Bluetooth activity. Please advise.

**Reply by Navid200** [reply]
What is MIUI Global 13.0.4?
Can you provide a link?

**Reply by dar-kod** [reply]
It is an Android based operating system, a modified Android ROM. The version which came with the phone is MIUI 12.5, but I upgraded to 13.0.2 already before. If I remember good, I had also disconnections in that time.

**Comment by dar-kod** [comment]
After 2 or 3 days, constantly temporary changing transmitter ID, it was started to work. But after change of transmitter I had again disconnections and I had to install BYODA. Please, if you solve this problem, inform me.

**Comment by dar-kod** [comment]
Here's how it might be possible to reproduce the Bluetooth disconnection with the Dexcom G6: force stop xDrip and reboot the phone. The problem also occurs when using the Cleaner application, which cleans cache files.

The only way that works for me is: Settings -> Apps -> Manage apps -> Bluetooth -> Clear data -> Clear cache. You should immediately reset the phone and in xDrip in Settings -> Dexcom Transmitter ID -> change the ID, click on OK, then return the ID to the original ID. If the disconnection occurs again, the process should be repeated and after 2 or possibly 3 times the connection should begin to function normally.

I don't know if this will help to solve this software problem or if it is a MIUI OS issue. I currently have the MIUI 13.0.6 version (android 12)
and xDrip+ 5ea552c-2022.12.09 version.
